apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs

```diff
@@ -245,13 +191,59 @@ internal string DatabaseToDrop
+        /// InsertOrUpdate, or Insert, then the text should be '[spanner command type] [table name]'
+        /// such as 'INSERT MYTABLE'.  Must not be null or empty.
+        /// </remarks>
+        /// <param name="commandText">The full command text containing a query, DDL statement or insert/update/delete
         /// operation.  The given text will be parsed and validated. Must not be null.</param>
         /// <returns>A <see cref="SpannerCommandTextBuilder"/> whose <see cref="SpannerCommandType"/> is determined
         /// from parsing <paramref name="commandText"/>.</returns>
-        public static SpannerCommandTextBuilder FromCommandText(string commandText) =>
-            new SpannerCommandTextBuilder(GaxPreconditions.CheckNotNullOrEmpty(commandText, nameof(commandText)));
+        /// <exception cref="ArgumentException"><paramref name="commandText"/> is not formatted correctly.</exception>
+        public static SpannerCommandTextBuilder FromCommandText(string commandText)
+        {
+            GaxPreconditions.CheckNotNullOrEmpty(commandText, nameof(commandText));
+            commandText = commandText.Trim();
+            var commandSections = commandText.Split(' ');
+            if (commandSections.Length < 2)
+            {
+                throw new ArgumentException($"'{commandText}' is not a recognized Spanner command.", nameof(commandText));
+            }
+
+            string commandName = commandSections[0].ToUpperInvariant();
+            if (!s_commandToCommandType.TryGetValue(commandName, out var commandType))
+            {
+                throw new ArgumentException($"'{commandName}' is not a recognized Spanner command.", nameof(commandText));
+            }
+
+            string targetTable = null;
+
+            // Any extra validation or assignments go here... currently just for the parameter-based DML cases.
+            switch (commandType)
+            {
+                case SpannerCommandType.Insert:
+                case SpannerCommandType.InsertOrUpdate:
+                case SpannerCommandType.Delete:
+                case SpannerCommandType.Update:
+                    if (commandSections.Length != 2)
+                    {
+                        throw new InvalidOperationException(
+                            $"Spanner {commandName} commands are specified as '{commandName} <table>' with " +
+                            "parameters added to customize the command with filtering or updated values.");
+                    }
+                    targetTable = ValidateTableName(commandSections[1], nameof(commandText));
+                    // This just normalizes the command text.
+                    commandText = $"{commandName} {targetTable}";
+                    break;
+            }
+            return new SpannerCommandTextBuilder(commandText: commandText, type: commandType, targetTable: targetTable, extraStatements: null);
```

Comment:
nit: not sure it's worth giving the param-names for all args here.

Reply:
I might reduce it to just the null then.